<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1319</width>
    <height>704</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="sizeIncrement">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <property name="sizeConstraint">
     <enum>QLayout::SetNoConstraint</enum>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="SideWindow" name="sidePannel" native="true">
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>300</height>
         </size>
        </property>
        <property name="title">
         <string>3D View</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="GLWidget" name="OpenGL" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>300</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>24</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolBox" name="toolBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>250</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>2</number>
        </property>
        <widget class="QWidget" name="RunMotion">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>783</width>
           <height>110</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Run Motion Planning</string>
         </attribute>
         <widget class="QWidget" name="RunMotionPlan" native="true">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>0</y>
            <width>151</width>
            <height>91</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <layout class="QVBoxLayout" name="allButtonsLayout">
             <item>
              <widget class="QPushButton" name="pushButtonRun">
               <property name="text">
                <string>Run</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonStop">
               <property name="text">
                <string>Stop</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonReset">
               <property name="text">
                <string>Reset</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QCheckBox" name="checkBoxWithShortCut">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>120</y>
            <width>121</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>With Short Cut</string>
          </property>
         </widget>
         <widget class="QRadioButton" name="radioButtonDiff">
          <property name="geometry">
           <rect>
            <x>180</x>
            <y>10</y>
            <width>91</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Diffusion</string>
          </property>
         </widget>
         <widget class="QRadioButton" name="radioButtonPRM">
          <property name="geometry">
           <rect>
            <x>180</x>
            <y>50</y>
            <width>61</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>PRM</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonResetGraph">
          <property name="geometry">
           <rect>
            <x>600</x>
            <y>40</y>
            <width>91</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Reset Graph</string>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="ViewerSettings">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>783</width>
           <height>110</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Viewer Settings 1</string>
         </attribute>
         <widget class="QGroupBox" name="groupBoxScene">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>0</y>
            <width>261</width>
            <height>91</height>
           </rect>
          </property>
          <property name="title">
           <string>Scene</string>
          </property>
          <widget class="QCheckBox" name="checkBoxTiles">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Tiles</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxWalls">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>30</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Walls</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxFloor">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>60</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Floor</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxShadows">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>60</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Shadow</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxSmooth">
           <property name="geometry">
            <rect>
             <x>170</x>
             <y>30</y>
             <width>81</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>Smooth</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxAxis">
           <property name="geometry">
            <rect>
             <x>170</x>
             <y>60</y>
             <width>81</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>Axis</string>
           </property>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBoxMPlanning">
          <property name="geometry">
           <rect>
            <x>290</x>
            <y>0</y>
            <width>141</width>
            <height>91</height>
           </rect>
          </property>
          <property name="title">
           <string>Motion Planning</string>
          </property>
          <widget class="QCheckBox" name="checkBoxDrawGraph">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>111</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Draw Graph</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxDrawTraj">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>60</y>
             <width>87</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Draw Traj</string>
           </property>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBox3DModel">
          <property name="geometry">
           <rect>
            <x>450</x>
            <y>0</y>
            <width>161</width>
            <height>91</height>
           </rect>
          </property>
          <property name="title">
           <string>3D Model</string>
          </property>
          <widget class="QCheckBox" name="checkBoxBB">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>141</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Bounding Boxes</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxGhosts">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>60</y>
             <width>101</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Ghosts</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="checkBoxFilaire">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>60</y>
             <width>96</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>Filaire</string>
           </property>
          </widget>
         </widget>
        </widget>
        <widget class="QWidget" name="Viewersettings2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>783</width>
           <height>110</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Viewer Settings 2</string>
         </attribute>
         <widget class="QDoubleSpinBox" name="doubleSpinBoxLightZ">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>60</y>
            <width>61</width>
            <height>21</height>
           </rect>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="doubleSpinBoxLightX">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>0</y>
            <width>61</width>
            <height>21</height>
           </rect>
          </property>
         </widget>
         <widget class="QDoubleSpinBox" name="doubleSpinBoxLightY">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>61</width>
            <height>21</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonRestoreView">
          <property name="geometry">
           <rect>
            <x>470</x>
            <y>0</y>
            <width>113</width>
            <height>32</height>
           </rect>
          </property>
          <property name="text">
           <string>Restore View</string>
          </property>
         </widget>
         <widget class="QSlider" name="horizontalSliderLightX">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>0</y>
            <width>311</width>
            <height>22</height>
           </rect>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSlider" name="horizontalSliderLightY">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>30</y>
            <width>311</width>
            <height>22</height>
           </rect>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSlider" name="horizontalSliderLightZ">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>60</y>
            <width>311</width>
            <height>22</height>
           </rect>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>0</y>
            <width>21</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>X</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>30</y>
            <width>21</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_4">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>60</y>
            <width>21</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Z</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButtonRestoreLight">
          <property name="geometry">
           <rect>
            <x>470</x>
            <y>30</y>
            <width>113</width>
            <height>32</height>
           </rect>
          </property>
          <property name="text">
           <string>Restore Light</string>
          </property>
         </widget>
         <widget class="QCheckBox" name="checkBoxDrawLightSource">
          <property name="geometry">
           <rect>
            <x>480</x>
            <y>60</y>
            <width>141</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Draw Light Source</string>
          </property>
         </widget>
        </widget>
        <widget class="QWidget" name="page">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>783</width>
           <height>110</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Show Trajectory</string>
         </attribute>
         <widget class="QPushButton" name="pushButtonShowTraj">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>0</y>
            <width>141</width>
            <height>32</height>
           </rect>
          </property>
          <property name="text">
           <string>Show Trajectory</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="lineEdit">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>50</y>
            <width>51</width>
            <height>22</height>
           </rect>
          </property>
         </widget>
         <widget class="QScrollBar" name="horizontalScrollBar">
          <property name="geometry">
           <rect>
            <x>170</x>
            <y>50</y>
            <width>231</width>
            <height>20</height>
           </rect>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>50</y>
            <width>61</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Speed</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1319</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuEnvironement">
     <property name="title">
      <string>Environement</string>
     </property>
     <addaction name="actionNameOfEnv"/>
    </widget>
    <addaction name="actionOpen"/>
    <addaction name="actionCloseEnvironement"/>
    <addaction name="menuEnvironement"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Environement</string>
    </property>
    <addaction name="actionOpenScenario"/>
    <addaction name="actionSaveScenarion"/>
    <addaction name="separator"/>
    <addaction name="actionRobot"/>
    <addaction name="action3DViewer"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuCollisionCheker">
    <property name="title">
     <string>Collision Cheker</string>
    </property>
    <addaction name="actionKCDPropietes"/>
   </widget>
   <widget class="QMenu" name="menuTrajectory">
    <property name="title">
     <string>Trajectory</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
   <addaction name="menuTrajectory"/>
   <addaction name="menuCollisionCheker"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionRobot">
   <property name="text">
    <string>Robot</string>
   </property>
  </action>
  <action name="action3DViewer">
   <property name="text">
    <string>3D-Viewer</string>
   </property>
  </action>
  <action name="actionKCDPropietes">
   <property name="text">
    <string>KCD Propietes</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionCloseEnvironement">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionOpenScenario">
   <property name="text">
    <string>Load Scenario</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSaveScenarion">
   <property name="text">
    <string>Save Scenario</string>
   </property>
  </action>
  <action name="actionNameOfEnv">
   <property name="text">
    <string>NameOfEnv</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>../qtOpenGL/glwidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SideWindow</class>
   <extends>QWidget</extends>
   <header>./sidewindow.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
